### ISSUE TITLE
e: Could not load module <Error module>

### ISSUE THREAD

**Issue by argenkiwi** [michaelbull/kotlin-result#85]
I am getting an obscure error message when attempting to build my Android application after refactoring a Repository interface to return Result: `e: Could not load module <Error module>`

It's so vague that I can't think of where to look for what could be causing the error. I found this [thread](https://youtrack.jetbrains.com/issue/KT-52757/Type-inference-for-builders-fails-if-inferred-from-a-function) which suggests the error is caused by type inheritance failing.

**Comment by michaelbull** [comment]
Is this an issue with kotlin-result or the kotlin compiler? I'm unsure as to how we would progress this issue with no stacktrace nor reproducability example.

**Comment by 05nelsonm** [comment]
Java9 Modules? Are you using with JavaFX? if so, you may have to add your own `module-info.java` file with the appropriate exports.

**Comment by argenkiwi** [comment]
Feel free to close the issue. All I could get from the logs is in the title of the issue. If I have the time to narrow it down further and it turns out to be an issue related to this library I will raise a new issue. 

I just thought I would raise awareness of it because, from my perspective as someone trying the library out for the first time, I am worried some of us could be experiencing this obscure issue and giving up on it early on, which would be a shame. 

Interestingly, the error is thrown by kaptGenerateStubsDebugKotlin for only 3 feature modules of the project which do not declare kotlin-result as a dependency. Maybe the issue is not directly related to the library but to the refactoring I had to do to make use of it. I managed to do a smaller change that makes use of this library but affects only one module and the issue did not occur.

**Comment by argenkiwi** [comment]
One more update: I went through the same systematic refactoring, which entailed replacing the return type of some methods of my Repository interface with, in this case, Arrow's Either type and I encountered the exact same issue. We can say this is definitely not a problem that is exclusive to kotlin-result. 

I am still unsure what the exact problem is, but I suspect it is a bug with [type inference](https://youtrack.jetbrains.com/issue/KT-52757/Type-inference-for-builders-fails-if-inferred-from-a-function) and perhaps the dependency injection library used in my project (Hilt/Dagger).
